cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -405,13 +410,27 @@ def _set_encrypted_service(self, service_type, alias, encrypted):
         service_path = Path(
             f"{self.global_config_dir}/services/{service_type}/{alias}.service"
         )
+        if not service_path.parent.is_dir():
+            service_path.parent.mkdir()
         with open(service_path, "wb") as f:
             f.write(encrypted)
 
     def _get_service(self, service_type, alias):
+        ConfigClass = ServiceConfig
+        if "class_path" in self.project_config.config["services"][service_type]:
+            ConfigClass = import_class(
+                self.project_config.services[service_type]["class_path"]
+            )
+        extra = []
+        # MarketingCloudServiceConfig requires a keychain
+        # so that it can load its configured oauth-client service
+        if service_type == "marketing-cloud":
+            extra = [self]
```

Comment:
Since you added the keychain param to ServiceConfig.__init__, I think you can always pass it and not need this special case.